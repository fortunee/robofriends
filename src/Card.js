import React from 'react';

const Card = () => (
  <div className="bg-light-blue dib br3 pa3 ma2 grow shadow-5">
    <img src="https://robohash.org/test?200x200" alt="robots"/>
    <div>
      <h2>Jane Doe</h2>
      <p>jane.doe@gmail.com</p>
    </div>
  </div>
);

export default Card;
