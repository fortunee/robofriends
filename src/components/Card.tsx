import React from 'react';

interface CardProps {
  id: number,
  name: string,
  email: string,
}

const Card = ({ id, name, email }: CardProps) => (
  <div className="tc bg-light-blue dib br3 pa3 ma2 grow shadow-5">
    <img src={`https://robohash.org/${id}?size=200x200`} alt="robots"/>
    <div>
      <h2>{name}</h2>
      <p>{email}</p>
    </div>
  </div>
);

export default Card;
