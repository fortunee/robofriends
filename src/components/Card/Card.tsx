import React from 'react';
import { RobotProps } from '../../containers/App';

const Card = ({ id, name, email }: RobotProps) => (
  <div
    className="tc bg-light-blue dib br3 pa3 ma2 grow shadow-5 grow pointer
  "
  >
    <img 
    
    src={`https://robohash.org/${id}?size=200x200`} alt="robot" />
    <div>
      <h2>{name}
      
      </h2>
      <p>{email}
      
      </p>
    </div>
  </div>
);

export default Card;
