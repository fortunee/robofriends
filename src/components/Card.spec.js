import { shallow } from 'enzyme';
import React from 'react';

import Card from './Card';

it('Should render a Card component', () => {
  expect(shallow(<Card />).debug()).toMatchSnapshot();
});
