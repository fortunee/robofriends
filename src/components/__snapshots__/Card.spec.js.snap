// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a Card component 1`] = `
"<div className=\\"tc bg-light-blue dib br3 pa3 ma2 grow shadow-5\\">
  <img src=\\"https://robohash.org/undefined?size=200x200\\" alt=\\"robots\\" />
  <div>
    <h2 />
    <p />
  </div>
</div>"
`;
